### PR TITLE
Add MyAnimeList anime data client

### LINKED ISSUE #2: Create an API client for myanimelist

### PR DESCRIPTION
**Pull request by voxlol**
Resolves #2

### CODE REVIEW

File: .env.local [hunk 1]
```diff
@@ -0,0 +1,2 @@
+MAL_USERNAME=
```
Comment (voxlol) [inline .env.local]: This file needs the other parameters in the `.env` file living on the c9 instance

File: data.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+                    const result = response.shift();
+                    resolve(result === null ? result : this._constructAnimeModel(result));
+                })
+                .catch(err => reject(err));
+        });
+    }
+
+    _constructAnimeModel(animeResponse) {
+        try {
+            const title = animeResponse.title;
```
Comment (voxlol) [inline data.js]: Is there a better way to checking the `animeResponse` has these properties on it? HasOwnProperty??

File: data.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    getAnime(name) {
+        throw "Implementation error";
+    }
+
+    _constructAnimeModel(animeResponse) {
+        throw "Implementation error";
+    }
+}
+
+class Anime {
```
Comment (voxlol) [inline data.js]: this should go to its own file eventually

File: data.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    constructor(params) {
+        this._validateParams(params);
```
Comment (voxlol) [inline data.js]: It might be worth making an abstract model class with `_validateParams` function signature

**Comment by theevenprime2** [comment]
LGTM 👍

**Pull request merged**